craft_application/commands/init.py

```diff
@@ -0,0 +1,144 @@
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Command to initialise a project."""
+
+from __future__ import annotations
+
+import argparse
+import importlib.resources
+import pathlib
+from textwrap import dedent
+from typing import cast
+
+import craft_cli
+
+from craft_application.util import humanize_list
+
+from . import base
+
+
+class InitCommand(base.AppCommand):
+    """Command to create initial project files.
+
+    The init command should always produce a working and ready-to-build project.
+    """
+
+    name = "init"
+    help_msg = "Create an initial project filetree"
+    overview = dedent(
+        """
+        Initialise a project.
+
+        If '<project-dir>' is provided, initialise in that directory,
+        otherwise initialise in the current working directory.
+
+        If '--name <name>' is provided, the project will be named '<name>'.
+        Otherwise, the project will be named after the directory it is initialised in.
+
+        '--profile <profile>' is used to initialise the project for a specific use case.
+
+        Init can work in an existing project directory. If there are any files in the
+        directory that would be overwritten, then init command will fail.
+        """
+    )
+    common = True
+
+    default_profile = "simple"
+    """The default profile to use when initialising a project."""
+
+    def fill_parser(self, parser: argparse.ArgumentParser) -> None:
+        """Specify command's specific parameters."""
+        parser.add_argument(
+            "project_dir",
+            type=pathlib.Path,
+            nargs="?",
+            default=None,
+            help="Path to initialise project in; defaults to current working directory.",
+        )
+        parser.add_argument(
+            "--name",
+            type=str,
+            default=None,
+            help="The name of project; defaults to the name of <project_dir>",
+        )
+        # TODO: this fails to render in `--help` (#530)
+        parser.add_argument(
+            "--profile",
+            type=str,
+            choices=self.profiles,
+            default=self.default_profile,
+            help=(
+                f"Use the specified project profile (default is {self.default_profile}, "
+                f"choices are {humanize_list(self.profiles, 'and')})"
+            ),
+        )
+
+    @property
+    def parent_template_dir(self) -> pathlib.Path:
```

Comment:
This is here rather than the service because we need the profiles before starting services, right?

Reply:
Correct